ray-operator/main.go

```diff
@@ -117,15 +119,29 @@ func main() {
 		setupLog.Info("Feature flag enable-batch-scheduler is enabled.")
 	}
 
-	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+	watchNamespaces := strings.Split(watchNamespace, ",")
+	options := ctrl.Options{
 		Scheme:                 scheme,
 		MetricsBindAddress:     metricsAddr,
 		Port:                   9443,
 		HealthProbeBindAddress: probeAddr,
 		LeaderElection:         enableLeaderElection,
 		LeaderElectionID:       "ray-operator-leader",
-		Namespace:              watchNamespace,
-	})
+	}
+
+	if len(watchNamespaces) == 1 {
+		options.Namespace = watchNamespaces[0]
+		if watchNamespaces[0] == "" {
```

Comment:
Based on [this doc](https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.14.6/pkg/manager#Options), if Namespace is specified, restricts the manager's cache to watch objects in the desired namespace. If the field is not specified, it defaults to all namespaces.